base/tests/unittests/com/google/idea/blaze/base/projectview/parser/ProjectViewParserTest.java

```diff
@@ -451,23 +452,24 @@ public void testListParserAcceptsWhitespace() throws Exception {
     assert projectViewFile != null;
     ProjectView projectView = projectViewFile.projectView;
 
-    assertThat(projectView)
-        .isEqualTo(
-            ProjectView.builder()
-                .add(
+    ProjectView expectedProjectView = ProjectView.builder()
+            .add(
                     ListSection.builder(DirectorySection.KEY)
-                        .add(DirectoryEntry.include(new WorkspacePath("dir0")))
-                        .add(TextBlock.of("  "))
-                        .add(TextBlock.of(""))
-                        .add(DirectoryEntry.include(new WorkspacePath("dir1")))
-                        .add(TextBlock.of("  ", "  "))
-                        .add(TextBlock.of("# comment"))
-                        .add(DirectoryEntry.include(new WorkspacePath("dir2")))
-                        .add(TextBlock.of(""))
-                        .add(TextBlock.of("  # commented out dir"))
-                        .add(TextBlock.of("  ")))
-                .add(TextBlockSection.of(TextBlock.of("# comment", "# comment")))
-                .build());
+                            .setFirstLineNumber(0)
+                            .add(DirectoryEntry.include(new WorkspacePath("dir0")), 1)
+                            .add(TextBlock.of(2, "  "))
+                            .add(TextBlock.of(3, ""))
+                            .add(DirectoryEntry.include(new WorkspacePath("dir1")), 4)
+                            .add(TextBlock.of(5, "  ", "  "))
+                            .add(TextBlock.of(7, "# comment"))
+                            .add(DirectoryEntry.include(new WorkspacePath("dir2")), 8)
+                            .add(TextBlock.of(9, ""))
+                            .add(TextBlock.of(10, "  # commented out dir"))
+                            .add(TextBlock.of(11, "  ")))
+            .add(TextBlockSection.of(TextBlock.of(12, "# comment", "# comment")))
```

Comment:
This test doesn't contain any list sections (I think). could you add a test that tests that lines are parsed correctly in all the sections you modified?

Reply:
It does cover the `ListSection` since the declarations are `Builder<T> setFirstLineNumber(int firstLineNumber)`, `public static <T> Builder<T> builder(SectionKey<T, ListSection<T>> sectionKey)`, and `public static final SectionKey<DirectoryEntry, ListSection<DirectoryEntry>> KEY =  SectionKey.of("directories");` which ultimately calls `ListSection` and checks it. In fact, if you build my Target Manager diff which already has all the code here you will see that even if you add duplicate directories and sections the correct one would be toggled when you click on the checkbox.